src/common.cpp

```diff
@@ -1993,3 +1994,28 @@ void redirect_tty_output() {
     show_stackframe(L'E', 99, 1);
     abort();
 }
+
+/// Return a path to a directory where we can store temporary files.
+wcstring get_path_to_tmp_dir()
+{
+    env_var_t env_tmpdir = env_get_string(L"TMPDIR");
+    if (!env_tmpdir.missing()) return env_tmpdir;
+
+#ifdef _CS_DARWIN_USER_TEMP_DIR
+    char osx_tmpdir[PATH_MAX];
+    if(confstr(_CS_DARWIN_USER_TEMP_DIR, osx_tmpdir, PATH_MAX - 1))
+    {
+        return str2wcstring(osx_tmpdir);
+    }
+#endif
+
+#ifdef P_tmpdir
```

Comment:
To avoid the compiler and lint tools complaining about unreachable code this should be structured as 

```
#ifdef P_tmpdir
    return P_tmpdir
#else
#ifdef _PATH_TMP_
    return _PATH_TMP_
#else
    return /tmp
#endif
#endif
```